project/AkkaDisciplinePlugin.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka
+
+import sbt._
+import Keys.{scalacOptions, _}
+import sbt.plugins.JvmPlugin
+
+/**
+  * Initial tests found:
+  * `akka-actor` 151 errors with `-Xfatal-warnings`, 6 without the flag
+  */
+object AkkaDisciplinePlugin extends AutoPlugin with ScalafixSupport {
+
+  import scoverage.ScoverageKeys.{coverageFailOnMinimum, coverageHighlighting, coverageMinimum}
+  import scalafix.sbt.ScalafixPlugin
+
+  override def trigger: PluginTrigger = allRequirements
+  override def requires: Plugins = JvmPlugin && ScalafixPlugin
+
+  override lazy val projectSettings = disciplineSettings
+
+  lazy val scalaFixSettings = Seq(
+    Compile / scalacOptions += "-Yrangepos")
+
+  lazy val scoverageSettings = Seq(
+    coverageMinimum := 60, // TODO set during review, may need tuning per module as well
+    coverageFailOnMinimum := false,
+    coverageHighlighting := true)
+
+  lazy val disciplineSettings =
+    scalaFixSettings ++
+      scoverageSettings ++ Seq(
+      Compile / scalacOptions ++= disciplineScalacOptions,
+      Compile / scalacOptions --= undisciplineScalacOptions,
+      Compile / console / scalacOptions --= Seq("-deprecation", "-Xfatal-warnings", "-Xlint", "-Ywarn-unused:imports"),
+      Compile / scalacOptions --= (CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((2, 13)) =>
+          Seq("-Ywarn-inaccessible", "-Ywarn-infer-any", "-Ywarn-nullary-override", "-Ywarn-nullary-unit", "-Ypartial-unification", "-Yno-adapted-args")
+        case Some((2, 12)) =>
+          Nil
+        case Some((2, 11)) =>
+          Seq("-Ywarn-extra-implicit", "-Ywarn-unused:_")
+        case _             =>
+          Nil
+      }))
+
+  /**
+    * Remain visibly filtered for future code quality work and removing.
+    */
+  val undisciplineScalacOptions = Seq(
+    "-Ywarn-value-discard",
+    "-Ywarn-numeric-widen",
+    "-Yno-adapted-args",
+    "-Xfatal-warnings")
+
+  /** Optimistic, this is desired over time.
+    * -Xlint and -Ywarn-unused: are included in commonScalaOptions.
+    * If eventually all modules use this plugin, we could migrate them here.
+    */
+  val disciplineScalacOptions = Seq(
+    // start: must currently remove, version regardless
+    "-Xfatal-warnings",
+    "-Ywarn-value-discard",
+    "-Yno-adapted-args",
+    "-Ywarn-numeric-widen",
+    // end
+    "-Xcheckinit",
+    "-Xfuture",
```

Comment:
This is "Turn on future language features", do we need that?

Reply:
Need? no but recommended, e.g. https://tpolecat.github.io/2017/04/25/scalac-flags.html